examples/slurm/hf/t5/pippy_t5.py

```diff
@@ -97,21 +97,76 @@ def trace(self, root, concrete_args=None):
         return graph
 
 
-def add_split_points(t5, sm_cnt):
-    num_encoders = 6  # encoders
-    num_lm_head = 1  # lm head
-    total_layers = t5.config.num_decoder_layers + num_encoders + num_lm_head
-    layers_per_rank = total_layers // sm_cnt
-    residual = total_layers % sm_cnt
-    split_after_layer = [ (i * layers_per_rank + (i if i < residual else residual)) for i in range(0, sm_cnt) ]
-    split_after_decoder = [ x - num_encoders for x in split_after_layer ]
-    for i in range(1, sm_cnt):
-        annotate_split_points(t5, {f'decoder.block.{split_after_decoder[i]}': PipeSplitWrapper.SplitPoint.BEGINNING})
+def add_split_points(t5, num_submodules):
+    if num_submodules == 1:
+        pass
+    elif num_submodules == 3:
+        # assert num_submodules == _add_split_points(t5, [16, 30])
+        assert num_submodules == _add_split_points(t5, [17, 31])
+    elif num_submodules == 4:
+        assert num_submodules == _add_split_points(t5, [13, 24, 35])
+    elif num_submodules == 7:
+        # assert num_submodules == _add_split_points(t5, [8, 14, 20, 26, 32, 38])
+        assert num_submodules == _add_split_points(t5, [9, 15, 21, 27, 33, 39])
+    elif num_submodules == 8:
+        # assert num_submodules == _add_split_points(t5, [7, 13, 19, 25, 31, 37, 43])
+        assert num_submodules == _add_split_points(t5, [9, 14, 19, 24, 29, 34, 39])
+    elif num_submodules == 15:
+        # assert num_submodules == _add_split_points(t5, [3, 6, 9, 12, 15, 18, 21, 24, 27, 30, 33, 36, 39, 42])
+        assert num_submodules == _add_split_points(t5, [1, 5, 9, 12, 15, 18, 21, 24, 27, 30, 33, 36, 39, 42])
+    elif num_submodules == 16:
+        # assert num_submodules == _add_split_points(t5, [3, 6, 9, 12, 15, 18, 21, 24, 27, 30, 33, 36, 39, 42, 44])
+        # assert num_submodules == _add_split_points(t5, [1, 4, 7, 8, 11, 14, 17, 20, 23, 26, 29, 32, 35, 38, 41])
+        assert num_submodules == _add_split_points(t5, [1, 4, 7, 10, 13, 16, 19, 22, 25, 28, 31, 34, 37, 40, 43])
+    else:
+        raise ValueError(f'Unsupported num_submodules = {num_submodules}')
+
```

Comment:
Nit: for some of the above arrays to be more readable, is it possible to use a `range(start, end, step)` coding?
For example: nStage 7, nStage 8, nStage 16 can use that.